src/axom/core/examples/core_numerics.cpp

```diff
@@ -0,0 +1,267 @@
+#include "axom/core/utilities/Utilities.hpp"
+
+#ifdef WIN32
+#include "windows.h"
+void sleep(int numSeconds)
+{
+  int numMilliSecs = numSeconds * 1000;
+  Sleep(numMilliSecs);
+}
+#else
+#include <unistd.h> // for sleep()
+#endif
+
+// C/C++ includes
+#include <iostream>
+#include <vector>
+
+void demoVectorOps()
+{
+  // _vecops_start
+  using namespace axom::numerics;
+
+  // First and second vectors
+  double u[] = {4., 1., 0.};
+  double v[] = {1., 2., 3.};
+  double w[] = {0., 0., 0.};
+  
+  std::cout << "Originally, u and v are" << std::endl <<
+    "u = [" << u[0] << ", " << u[1] << ", " << u[2] << "]" << std::endl <<
+    "v = [" << v[0] << ", " << v[1] << ", " << v[2] << "]" << std::endl;
+
+  // Calculate dot and cross products
+  double dotprod = dot_product(u, v, 3);
+  cross_product(u, v, w);
+
+  std::cout << "The dot product is " << dotprod <<
+    " and the cross product is" << std::endl <<
+    "[" << w[0] << ", " << w[1] << ", " << w[2] << "]" << std::endl;
+
+  // Make u orthogonal to v, then normalize v
+  make_orthogonal(u, v, 3);
+  normalize(v, 3);
+
+  std::cout << "Now orthogonal u and normalized v are" << std::endl <<
+    "u = [" << u[0] << ", " << u[1] << ", " << u[2] << "]" << std::endl <<
+    "v = [" << v[0] << ", " << v[1] << ", " << v[2] << "]" << std::endl;
+
+  // Fill a linear space
+  const int lincount = 45;
+  double s[lincount];
+  linspace(1., 9., s, lincount);
+
+  // Find the real roots of a cubic equation.
+  // (x + 2)(x - 1)(2x - 3) = 0 = 2x^3 - x^2 - 7x + 6 has real roots at
+  // x = -2, x = 1, x = 1.5.
+  double coeff[] = {6., -7., -1., 2.};
+  double roots[3];
+  int numRoots;
+  int result = solve_cubic(coeff, roots, numRoots);
+
+  std::cout << "Root-finding returned " << result << " (should be 0, success)."
+    "  Found " << numRoots << " roots (should be 3)" << std::endl <<
+    "at x = " << roots[0] << ", " << roots[1] << ", " << roots[2] <<
+    " (should be x = -2, 1, 1.5 in arbitrary order)." << std::endl;
+  // _vecops_end
+}
+
+void display_eigs(double * eigvec, double * eigval, int nrows, int i)
+{
+  // display eigen vectors and values
+  std::cout << "Eigenvalue " << i << " = " << eigval[i] << " Eigenvector " <<
+    i << " = [";
+  for (int j = 0; j < nrows; ++j)
+  {
+    if (j > 0) std::cout << ", ";
+    std::cout << eigvec[i * nrows + j];
+  }
+  std::cout << "]" << std::endl;
+}
+
+void display_eigs(axom::numerics::Matrix< double > & eigvec, 
+                  double * eigval, 
+                  int i)
+{
+  double * p_eigvec = eigvec.data();
+  int nrows = eigvec.getNumRows();
+  display_eigs(p_eigvec, eigval, nrows, i);
+}
+
+void demoMatrix()
+{
+  // _matctor_start
+  using namespace axom::numerics;
+
+  // Here's a 3X3 matrix of double values, initialized from an array.
+  const int nrows = 3;
+  const int ncols = 3;
+  double val[9] = {0.6, 2.4, 1.1, 2.4, 0.6, -.1, 1.1, -.1, 0.6};
+  Matrix< double > A(nrows, ncols, val, true);
+  
```

Comment:
It might be worthwhile adding a snippet to show how to extract a row or column from `Matrix` using `getRow()` and `getColumn()` since the signatures are different -- Both return a pointer, but `getRow()` requires extra `out` parameters to get the stride.

Reply:
Done.